api/api-agents-v5/src/main/java/com/thoughtworks/go/apiv5/agents/AgentsControllerV5.java

```diff
@@ -104,8 +107,14 @@ public String index(Request request, Response response) throws IOException {
 
     public String show(Request request, Response response) throws IOException {
         final AgentInstance agentInstance = fetchEntityFromConfig(request.params("uuid"));
+        List<String> environments;
+        List<EnvironmentConfig> environmentConfigs = Collections.emptyList();
+        if (!agentInstance.agentConfig().getEnvironments().isEmpty()) {
+            environments = Arrays.asList(agentInstance.agentConfig().getEnvironments().split(","));
+            environments.forEach(environmentName -> environmentConfigs.add(environmentConfigService.named(environmentName)));
+        }
```

Comment:
IMO, it would be better if the comma-separated representation of the environments is hidden away inside the `AgentConfig` or `Agent` object. We would then be able to call `agentConfig.getEnvironments()` that returns a `List<String>`

Reply:
Refactored. The `Agent.getEnvironmentsAsList` will return a list of env name or an empty list.